### PR TITLE
[BISERVER-14192] - ASCII Control Characters are being allowed in Fold…

### PR DESCRIPTION
**Pull request by bantonio82**
…ers and File Names causing repository corruption when backing up and restoring

This PR needs the code from the PR https://github.com/pentaho/pentaho-commons-gwt-modules/pull/760, so please merge it before this.

@pentaho-lmartins @ssamora

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 49m 48s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1381**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4627/1/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java [hunk 1]
```diff
@@ -1916,9 +1918,14 @@ public RepositoryFileTreeDto doGetTree( @PathParam ( "pathId" ) String pathId, @
   @Produces ( { MediaType.WILDCARD } )
   @StatusCodes( {
     @ResponseCode( code = 200, condition = "Successfully renamed file." ),
-    @ResponseCode( code = 200, condition = "File to be renamed does not exist." ) } )
+    @ResponseCode( code = 200, condition = "File to be renamed does not exist." ),
```
Comment (pentaho-lmartins) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java]: Code 200 is repeated.

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java [hunk 2]
```diff
@@ -1916,9 +1918,14 @@ public RepositoryFileTreeDto doGetTree( @PathParam ( "pathId" ) String pathId, @
...
+    @ResponseCode( code = 403, condition = "Forbidden." ) } )
```
Comment (pentaho-lmartins) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java]: Adding new response codes will require updating documentation of this API.

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java [hunk 3]
```diff
@@ -2000,6 +2007,7 @@ public Response doSetMetadata( @PathParam ( "pathId" ) String pathId, List<Strin
   @Consumes ( { MediaType.WILDCARD } )
   @StatusCodes ( {
       @ResponseCode ( code = 200, condition = "Successfully created folder." ),
+      @ResponseCode ( code = 403, condition = "Forbidden." ),
```
Comment (pentaho-lmartins) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java]: Same here.

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/FileService.java [hunk 4]
```diff
@@ -1660,6 +1669,10 @@ public String idToPath( String pathId ) {
     return FileUtils.idToPath( pathId );
   }
 
+  public boolean containsControlCharacters( String pathId ) {
```
Comment (pentaho-lmartins) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/FileService.java]: Why are we adding this method just to call another method?

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/utils/FileUtils.java [hunk 5]
```diff
@@ -85,6 +87,22 @@ public static boolean containsReservedCharacter( String path, char[] reserved )
...
+   *
+   * @param path to be validated
+   * @return {@code true} if any of {@code Control Characters} is contained in {@code path}
+   */
+
+  public static boolean containsControlCharacters( String path ) {
+    if ( StringUtils.isNotBlank( path ) ) {
+      Pattern pattern = Pattern.compile( "[\\x00-\\x1F\\x7F]" );
+      Matcher matcher = pattern.matcher( path );
+      return matcher.find();
```
Comment (pentaho-lmartins) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/utils/FileUtils.java]: Why not just:
```
      RegExp pattern = RegExp.compile( "[\\x00-\\x1F\\x7F]" );
      return pattern.test( path );
```

**Comment by pentaho-lmartins** [comment]
This doesn't build....
Can you please see why?

**Comment by buildguy** [comment]
### :x: Build failed in 1m 18s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 24m 18s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1489**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4627/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**